tests/realtikvtest/txntest/txn_test.go

```diff
@@ -595,3 +595,42 @@ func TestDMLWithAddForeignKey(t *testing.T) {
 
 	require.True(t, errDML != nil || errDDL != nil)
 }
+
+func TestLockKeysInInsertIgnore(t *testing.T) {
+	// FIXME: this test will fail in the unistore. After fixing the issue in unistore, please move
+	// this test to `foreign_key_test.go`.
```

Comment:
I still don't know why this case will fail in the unistore. Maybe the transaction implementation has some bugs (or just different behaviors).

I tried this case with/without `IGNORE`. It seems that the foreign key behaves strangely in unistore :thinking: :


Prepare:

```sql
create table parent (id int primary key);
create table child(id int primary key, ref int, foreign key (ref) references parent(id));

insert into parent values (1);
```

Execute:

| Transaction 1    | Transaction 2 |
| -------- | ------- |
| BEGIN;  | BEGIN;    |
| INSERT INTO child VALUES (1, 1); |     |
|     | UPDATE parent SET id = 2 WHERE id = 1    |
| COMMIT; | COMMIT; |

Reply:
Maybe it should get a snapshot with `ForUpdateTS`?

Reply:
I record this issue in https://github.com/pingcap/tidb/issues/56663.

Reply:
It'll be fixed in https://github.com/pingcap/tidb/pull/56821